### ISSUE TITLE
[ZLIB] Requires debug suffix libd or libd1 on linux

### ISSUE THREAD

**Issue by Neumann-A** [microsoft/vcpkg#6644]
Reason:

```
set(ZLIB_NAMES z zlib zdll zlib1)
set(ZLIB_NAMES_DEBUG zlibd zlibd1)
```

from findZLIB in CMake modules. Debug and Release libraries are both called libz.a under linux

**Comment by JackBoosY** [comment]
@Neumann-A Maybe we not only need to modify the generated name, but also need to modify findzlib,cmake?

**Comment by Neumann-A** [comment]
I will probably open a CMake bug report for this one becaus zlibstatic is also missing from the list. I dont think people will care too much about how a debug library is actually named.

**Comment by Neumann-A** [comment]
related upstream MR: https://gitlab.kitware.com/cmake/cmake/merge_requests/3384

**Comment by Neumann-A** [comment]
seems like it will be change in cmake 3.15

**Comment by JackBoosY** [comment]
This issue is still not resolved in 1.2.11.

**Comment by Neumann-A** [comment]
the wrapper in zlib should fix this.
